File: python/0094-binary-tree-inorder-traversal.py
from typing import List, Optional

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []

        def dfs(currNode, res):
            if not currNode:
                return
            dfs(currNode.left, res)
            res.append(currNode.val)
            dfs(currNode.right, res)
        
        dfs(root, res)
        return res
        

File: java/0094-binary-tree-inorder-traversal.java

// Definition for a binary tree node.

import java.util.ArrayList;
import java.util.List;

public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        dfs(res, root);
        return res;
    }

    public void dfs(List<Integer> res, TreeNode currNode) {
        if (currNode == null) {
            return;
        }
        dfs(res, currNode.left);
        res.add(currNode.val);
        dfs(res, currNode.right);
        return;
    }
}